# DP: ld/testsuite/ld-unique: Fix running unique tests on ARM

The @ character is a comment character on ARM, so use % instead. Also
use a wider glob for matching ARM targets to make sure the test gets
run.

ld/testsuite/ChangeLog:

2014-10-23  Will Newton  <will.newton@linaro.org>

	* ld-unique/unique.exp: Use a wider glob for matching ARM
	targets.
	* ld-unique/unique.s: Use % instead of @ in .type directive.
	* ld-unique/unique_shared.s: Likewise.
---
 ld/testsuite/ld-unique/unique.exp      | 2 +-
 ld/testsuite/ld-unique/unique.s        | 2 +-
 ld/testsuite/ld-unique/unique_shared.s | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ld/testsuite/ld-unique/unique.exp b/ld/testsuite/ld-unique/unique.exp
index a93f9b2..994ed87 100644
--- a/ld/testsuite/ld-unique/unique.exp
+++ b/ld/testsuite/ld-unique/unique.exp
@@ -28,7 +28,7 @@
 # arm, powerpc, and sparc so far.
 if {!(([istarget "i?86-*-*"]
        || [istarget "x86_64-*-*"]
-       || [istarget "arm-*-*"]
+       || [istarget "arm*-*-*"]
        || [istarget "powerpc*-*-*"]
        || [istarget "sparc*-*-*"])
       && ([istarget "*-*-elf*"]
diff --git a/ld/testsuite/ld-unique/unique.s b/ld/testsuite/ld-unique/unique.s
index 9b0593c..7477a69 100644
--- a/ld/testsuite/ld-unique/unique.s
+++ b/ld/testsuite/ld-unique/unique.s
@@ -1,4 +1,4 @@
-	.type a, @gnu_unique_object
+	.type a, %gnu_unique_object
 a:	.long 0
 	.size a, .-a
 
diff --git a/ld/testsuite/ld-unique/unique_shared.s b/ld/testsuite/ld-unique/unique_shared.s
index b18a5b1..8022291 100644
--- a/ld/testsuite/ld-unique/unique_shared.s
+++ b/ld/testsuite/ld-unique/unique_shared.s
@@ -1,3 +1,3 @@
-	.type b, @gnu_unique_object
+	.type b, %gnu_unique_object
 b:	.long 0
 	.size b, .-b
-- 
1.9.3

